packages/vite/src/node/optimizer/scan.ts

```diff
@@ -131,6 +131,10 @@ export async function scanImports(config: ResolvedConfig): Promise<{
 }
 
 function globEntries(pattern: string | string[], config: ResolvedConfig) {
+  if (Array.isArray(pattern)) {
+    // If a pattern is `undefined` then `glob` chokes.
+    pattern = pattern.filter(Boolean)
+  }
```

Comment:
I think comment is not really needed here, and you could simplify this like that:

```js
const globPreparedPattern = Array.isArray(pattern) ? pattern.filter(Boolean) : pattern;
```

You don't override function param and you can keep code in not mutable way. I hope it makes sense.

Reply:
I think the comment is good, so that it won't be reintroduced in future changes